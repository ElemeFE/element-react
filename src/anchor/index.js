import AnchorSmooth from './AnchorSmooth';
import GoTop from './GoTop';

export {AnchorSmooth,GoTop};