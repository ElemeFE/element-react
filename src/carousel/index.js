import Carousel from './Carousel';
import CarouselItem from './CarouselItem';

Carousel.Item = CarouselItem;

export default Carousel;
