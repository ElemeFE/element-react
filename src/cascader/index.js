import Cascader from './Cascader';

export default Cascader;
