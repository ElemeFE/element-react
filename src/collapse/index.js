import Collapse from './Collapse';
import CollapseItem from './CollapseItem';

Collapse.Item = CollapseItem;

export default Collapse;
