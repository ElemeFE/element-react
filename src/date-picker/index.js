/*
revision version: 
  48996c6658300b7f9f01b9b4011a531f17e30b75
*/
export { default as TimeSelect } from './TimeSelect'
export { default as TimePicker } from './TimePicker'
export { default as TimeRangePicker } from './TimeRangePicker'
export { default as DatePicker } from './DatePicker'
export { default as DateRangePicker } from './DateRangePicker'