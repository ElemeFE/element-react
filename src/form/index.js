import Form from './Form';
import FormItem from './FormItem';

Form.Item = FormItem;

export default Form;
