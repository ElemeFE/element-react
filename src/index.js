export { default as i18n } from './locale';

export { default as Alert } from './alert';
export { default as Button } from './button';
export { default as Card } from './card';
export { default as Layout } from './layout';
export { default as Loading } from './loading';
export { default as Message } from './message';
export { default as MessageBox } from './message-box';
export { default as Notification } from './notification';
export { default as Radio } from './radio';
export { default as Dialog } from './dialog';
export { default as Rate } from './rate';
export { default as Progress } from './progress';
export { default as Badge } from './badge';
export { default as Tabs } from './tabs';
export { default as Tree } from './tree';
export { default as Input } from './input';
export { default as Icon } from './icon';
export { default as Menu } from './menu';
export { default as Steps } from './steps';
export { default as Breadcrumb } from './breadcrumb';
export { default as Tooltip } from './tooltip';
export { default as InputNumber } from './input-number';
export { default as Checkbox } from './checkbox';
export { default as Slider } from './slider';
export { default as Table } from './table'
export { default as Switch } from './switch';
export { default as Form } from './form';
export { default as Upload } from './upload';
export { default as Tag } from './tag';
export { default as Select } from './select';
export { default as Dropdown } from './dropdown';
export { default as Popover } from './popover';
export { default as Pagination } from './pagination';
export { default as AutoComplete } from './auto-complete';
export { TimeSelect, TimePicker, TimeRangePicker, DatePicker, DateRangePicker } from './date-picker';
export { default as Carousel } from './carousel';
export { default as Collapse } from './collapse';
export { default as ColorPicker } from './color-picker';
export { default as Cascader } from './cascader';
export { default as Transfer } from './transfer';
