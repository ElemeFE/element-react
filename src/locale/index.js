import format from './format';
import defaultLang from './lang/en';

let _lang = defaultLang

function use(lang) {
  _lang = lang;
}

function t(path, options) {
  const array = path.split('.');
    let current = _lang;

    for (var i = 0, j = array.length; i < j; i++) {
      var property = array[i];
      var value = current[property];
      if (i === j - 1) return format(value, options);
      if (!value) return '';
      current = value;
    }
    return '';
}

export default {
  use,
  t
}
