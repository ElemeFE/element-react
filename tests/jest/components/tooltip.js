import React from 'react';
import { shallow, mount } from 'enzyme';
import { Button, Tooltip } from '../../../src';
import renderer from 'react-test-renderer';

test('Basic Usage', () => {
  const tooltip = renderer.create(
    <div className="box">
      <div className="top">
        <Tooltip className="item" effect="dark" content="Top Left prompts info" placement="top-start">
          <Button>top-start</Button>
        </Tooltip>
        <Tooltip className="item" effect="dark" content="Top Center prompts info" placement="top">
          <Button>top</Button>
        </Tooltip>
        <Tooltip className="item" effect="dark" content="Top Right prompts info" placement="top-end">
          <Button>top-end</Button>
        </Tooltip>
      </div>
      <div className="left">
        <Tooltip className="item" effect="dark" content="Left Top prompts info" placement="left-start">
          <Button>left-start</Button>
        </Tooltip>
        <Tooltip className="item" effect="dark" content="Left Center prompts info" placement="left">
          <Button>left</Button>
        </Tooltip>
        <Tooltip className="item" effect="dark" content="Left Bottom prompts info" placement="left-end">
          <Button>left-end</Button>
        </Tooltip>
      </div>
      <div className="right">
        <Tooltip className="item" effect="dark" content="Right Top prompts info" placement="right-start">
          <Button>right-start</Button>
        </Tooltip>
        <Tooltip className="item" effect="dark" content="Right Center prompts info" placement="right">
          <Button>right</Button>
        </Tooltip>
        <Tooltip className="item" effect="dark" content="Right Bottom prompts info" placement="right-end">
          <Button>right-end</Button>
        </Tooltip>
      </div>
      <div className="bottom">
        <Tooltip className="item" effect="dark" content="Bottom Left prompts info" placement="bottom-start">
          <Button>bottom-start</Button>
        </Tooltip>
        <Tooltip className="item" effect="dark" content="Bottom Center prompts info" placement="bottom">
          <Button>bottom</Button>
        </Tooltip>
        <Tooltip className="item" effect="dark" content="Bottom Right prompts info" placement="bottom-end">
          <Button>bottom-end</Button>
        </Tooltip>
      </div>
    </div>
  );

  expect(tooltip.toJSON()).toMatchSnapshot();
});

test('Theme', () => {
  const tooltip1 = shallow(
    <Tooltip content="Top center" placement="top">
      <Button>Dark</Button>
    </Tooltip>
  );

  expect(tooltip1.find('div .el-tooltip__popper').at(0).hasClass('is-dark')).toEqual(true);

  const tooltip2 = mount(
    <div>
      <Tooltip content="Bottom center" placement="bottom" effect="light">
        <Button>Light</Button>
      </Tooltip>
    </div>
  );

  expect(tooltip2.find('div .el-tooltip__popper').at(0).hasClass('is-light')).toEqual(true);

});

test('More Content', () => {
  const tooltip = mount(
    <Tooltip placement="top" content={<div>multiple lines<br/>second line</div>}>
      <Button>Top center</Button>
    </Tooltip>
  );

  expect(tooltip.contains(<div>multiple lines<br/>second line</div>)).toBe(true);
});

test('Advanced usage', () => {
  const state = {
    disabled: false
  };

  const tooltip = mount(
    <Tooltip disabled={ state.disabled } content="click to close tooltip function" placement="bottom" effect="light">
      <Button onClick={ e => {state.disabled = true}}>{`click to ${state.disabled ? 'active' : 'close' } tooltip function`}</Button>
    </Tooltip>
  );

  expect(state.disabled).toBe(false);
  expect(tooltip.toString()).toMatchSnapshot();

  tooltip.find('Button').simulate('click');

  expect(state.disabled).toBe(true);
  expect(tooltip.toString()).toMatchSnapshot();
});