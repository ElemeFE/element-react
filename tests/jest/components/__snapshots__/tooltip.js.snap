// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Advanced usage 1`] = `"[object Object]"`;

exports[`Advanced usage 2`] = `"[object Object]"`;

exports[`Basic Usage 1`] = `
<div
  className="box"
>
  <div
    className="top"
  >
    <div
      className="el-tooltip item"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={Object {}}
    >
      <div
        className="el-tooltip__rel"
      >
        <div>
          <button
            className="el-button el-button--default"
            disabled={false}
            onClick={[Function]}
            style={Object {}}
            type="button"
          >
            <span>
              top-start
            </span>
          </button>
        </div>
      </div>
      <span
        className={undefined}
        style={undefined}
      >
        <div
          className="el-tooltip__popper is-dark"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <div>
            Top Left prompts info
          </div>
          <div
            className="popper__arrow"
          />
        </div>
      </span>
    </div>
    <div
      className="el-tooltip item"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={Object {}}
    >
      <div
        className="el-tooltip__rel"
      >
        <div>
          <button
            className="el-button el-button--default"
            disabled={false}
            onClick={[Function]}
            style={Object {}}
            type="button"
          >
            <span>
              top
            </span>
          </button>
        </div>
      </div>
      <span
        className={undefined}
        style={undefined}
      >
        <div
          className="el-tooltip__popper is-dark"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <div>
            Top Center prompts info
          </div>
          <div
            className="popper__arrow"
          />
        </div>
      </span>
    </div>
    <div
      className="el-tooltip item"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={Object {}}
    >
      <div
        className="el-tooltip__rel"
      >
        <div>
          <button
            className="el-button el-button--default"
            disabled={false}
            onClick={[Function]}
            style={Object {}}
            type="button"
          >
            <span>
              top-end
            </span>
          </button>
        </div>
      </div>
      <span
        className={undefined}
        style={undefined}
      >
        <div
          className="el-tooltip__popper is-dark"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <div>
            Top Right prompts info
          </div>
          <div
            className="popper__arrow"
          />
        </div>
      </span>
    </div>
  </div>
  <div
    className="left"
  >
    <div
      className="el-tooltip item"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={Object {}}
    >
      <div
        className="el-tooltip__rel"
      >
        <div>
          <button
            className="el-button el-button--default"
            disabled={false}
            onClick={[Function]}
            style={Object {}}
            type="button"
          >
            <span>
              left-start
            </span>
          </button>
        </div>
      </div>
      <span
        className={undefined}
        style={undefined}
      >
        <div
          className="el-tooltip__popper is-dark"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <div>
            Left Top prompts info
          </div>
          <div
            className="popper__arrow"
          />
        </div>
      </span>
    </div>
    <div
      className="el-tooltip item"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={Object {}}
    >
      <div
        className="el-tooltip__rel"
      >
        <div>
          <button
            className="el-button el-button--default"
            disabled={false}
            onClick={[Function]}
            style={Object {}}
            type="button"
          >
            <span>
              left
            </span>
          </button>
        </div>
      </div>
      <span
        className={undefined}
        style={undefined}
      >
        <div
          className="el-tooltip__popper is-dark"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <div>
            Left Center prompts info
          </div>
          <div
            className="popper__arrow"
          />
        </div>
      </span>
    </div>
    <div
      className="el-tooltip item"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={Object {}}
    >
      <div
        className="el-tooltip__rel"
      >
        <div>
          <button
            className="el-button el-button--default"
            disabled={false}
            onClick={[Function]}
            style={Object {}}
            type="button"
          >
            <span>
              left-end
            </span>
          </button>
        </div>
      </div>
      <span
        className={undefined}
        style={undefined}
      >
        <div
          className="el-tooltip__popper is-dark"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <div>
            Left Bottom prompts info
          </div>
          <div
            className="popper__arrow"
          />
        </div>
      </span>
    </div>
  </div>
  <div
    className="right"
  >
    <div
      className="el-tooltip item"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={Object {}}
    >
      <div
        className="el-tooltip__rel"
      >
        <div>
          <button
            className="el-button el-button--default"
            disabled={false}
            onClick={[Function]}
            style={Object {}}
            type="button"
          >
            <span>
              right-start
            </span>
          </button>
        </div>
      </div>
      <span
        className={undefined}
        style={undefined}
      >
        <div
          className="el-tooltip__popper is-dark"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <div>
            Right Top prompts info
          </div>
          <div
            className="popper__arrow"
          />
        </div>
      </span>
    </div>
    <div
      className="el-tooltip item"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={Object {}}
    >
      <div
        className="el-tooltip__rel"
      >
        <div>
          <button
            className="el-button el-button--default"
            disabled={false}
            onClick={[Function]}
            style={Object {}}
            type="button"
          >
            <span>
              right
            </span>
          </button>
        </div>
      </div>
      <span
        className={undefined}
        style={undefined}
      >
        <div
          className="el-tooltip__popper is-dark"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <div>
            Right Center prompts info
          </div>
          <div
            className="popper__arrow"
          />
        </div>
      </span>
    </div>
    <div
      className="el-tooltip item"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={Object {}}
    >
      <div
        className="el-tooltip__rel"
      >
        <div>
          <button
            className="el-button el-button--default"
            disabled={false}
            onClick={[Function]}
            style={Object {}}
            type="button"
          >
            <span>
              right-end
            </span>
          </button>
        </div>
      </div>
      <span
        className={undefined}
        style={undefined}
      >
        <div
          className="el-tooltip__popper is-dark"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <div>
            Right Bottom prompts info
          </div>
          <div
            className="popper__arrow"
          />
        </div>
      </span>
    </div>
  </div>
  <div
    className="bottom"
  >
    <div
      className="el-tooltip item"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={Object {}}
    >
      <div
        className="el-tooltip__rel"
      >
        <div>
          <button
            className="el-button el-button--default"
            disabled={false}
            onClick={[Function]}
            style={Object {}}
            type="button"
          >
            <span>
              bottom-start
            </span>
          </button>
        </div>
      </div>
      <span
        className={undefined}
        style={undefined}
      >
        <div
          className="el-tooltip__popper is-dark"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <div>
            Bottom Left prompts info
          </div>
          <div
            className="popper__arrow"
          />
        </div>
      </span>
    </div>
    <div
      className="el-tooltip item"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={Object {}}
    >
      <div
        className="el-tooltip__rel"
      >
        <div>
          <button
            className="el-button el-button--default"
            disabled={false}
            onClick={[Function]}
            style={Object {}}
            type="button"
          >
            <span>
              bottom
            </span>
          </button>
        </div>
      </div>
      <span
        className={undefined}
        style={undefined}
      >
        <div
          className="el-tooltip__popper is-dark"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <div>
            Bottom Center prompts info
          </div>
          <div
            className="popper__arrow"
          />
        </div>
      </span>
    </div>
    <div
      className="el-tooltip item"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={Object {}}
    >
      <div
        className="el-tooltip__rel"
      >
        <div>
          <button
            className="el-button el-button--default"
            disabled={false}
            onClick={[Function]}
            style={Object {}}
            type="button"
          >
            <span>
              bottom-end
            </span>
          </button>
        </div>
      </div>
      <span
        className={undefined}
        style={undefined}
      >
        <div
          className="el-tooltip__popper is-dark"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <div>
            Bottom Right prompts info
          </div>
          <div
            className="popper__arrow"
          />
        </div>
      </span>
    </div>
  </div>
</div>
`;
